scipy/special/orthogonal.pyi

```diff
@@ -0,0 +1,319 @@
+from typing import (
+    Any,
+    Callable,
+    List,
+    Optional,
+    overload,
+    Tuple,
+    Union,
+)
+from typing_extensions import Literal
+
+import numpy
+
+_IntegerType = Union[int, numpy.integer]
+_FloatingType = Union[float, numpy.floating]
```

Comment:
It looks to me like these would also benefit from moving to `numpy-stubs`.

Reply:
Yeah, seems reasonable. Maybe the fact that we're already finding other types that would be useful to add is ammunition in favor of not putting types in the top-level NumPy namespace.

Reply:
Could you use `numbers` instead here?
```
_IntegerType = numbers.Integral
_FloatingType = numbers.Real
```

Reply:
Ah yes, so generally

https://github.com/python/mypy/issues/3186

has made this not work very well.